dts/arm/nucleo_f401re.dts

```diff
@@ -27,3 +27,31 @@
 	current-speed = <115200>;
 	status = "ok";
 };
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
```

Comment:
After discussing with Andy, this is OK -- partitions are defined as an offset / size, not address / size, in each reg.

But there is another bug elsewhere that needs to be addressed. This commit is OK though.